app/code/Bulbulatory/Recommendations/Model/ResourceModel/Table.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Model\ResourceModel;
+
+class Table extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb

```

Comment:
Resource model and model classes should be named after an entity which they are referring to - they refer to a table named `bulbulatory_recommendations` where each row is a single `Recommendation` so this should be a name of these classes. Same applies to a `Collection` class - it should be placed inside a `Recommendation` folder.

Naming these classes `Table` is not a mistake as the grid still works. But these classes will be used later to operate on recommendations and perform different actions on them so naming these classes after an entity makes it easier to develop in general.

Reply:
@TM18

> Resource model and model classes should be named after an entity which they are referring to - they refer to a table named bulbulatory_recommendations where each row is a single Recommendation so this should be a name of these classes. Same applies to a Collection class - it should be placed inside a Recommendation folder.
> 
> Naming these classes Table is not a mistake as the grid still works. But these classes will be used later to operate on recommendations and perform different actions on them so naming these classes after an entity makes it easier to develop in general.

Fixed!